sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuBroadcastHashJoinExecBase.scala

```diff
@@ -99,6 +123,36 @@ abstract class GpuBroadcastHashJoinExecBase(
     STREAM_TIME -> createNanoTimingMetric(DEBUG_LEVEL, DESCRIPTION_STREAM_TIME),
     JOIN_TIME -> createNanoTimingMetric(DEBUG_LEVEL, DESCRIPTION_JOIN_TIME))
 
+  override lazy val buildAttrList: List[Attribute] = if (postBuildCondition.isEmpty) {
+    buildPlan.output.toList
+  } else {
+    postBuildCondition.map(expr => expr.toAttribute)
+  }
```

Comment:
How is this different than postBuildAttr?  The naming is very similar and confusing.  I think some of the confusion here is that the join is only partially doing the AST fixup on its own.  It does the pre-project but only for the build side not the stream side, and does not do the post-project.  That requires additional workarounds in here that I think would go away if the join did all of the pre-projects and post project directly and there's never a project exec injected into the plan for the AST splitting.

Having the join deal with only part of the AST splitting makes it much harder to follow what's going on when focusing on the join node.  If the join node took the original left and right plans and emitted what it did before, taking an optional parameter (e.g. as a case class descriptor of the pre/post projects) or set of parameters to describe the pre-projects then it would be easier to follow since it's self-contained (and also likely to cause less issues since the join node is semantically equivalent to what it did before re: inputs/outputs).